internal/datanode/data_node.go

```diff
@@ -314,6 +316,13 @@ func (node *DataNode) handleWatchInfo(e *event, key string, data []byte) {
 			return
 		}
 
+		if watchInfo.State != datapb.ChannelWatchState_ToWatch && // start watch
+			watchInfo.State != datapb.ChannelWatchState_ToRelease && // start release
+			watchInfo.State != datapb.ChannelWatchState_Uncomplete { // legacy state, equal to ToWatch
```

Comment:
I can see the same code block appeared here and again below  (event_manager.go:118). Maybe worth making a function for them?